packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -0,0 +1,386 @@
+        keccak256(bytes("FederatedAttestations")),
+        keccak256("1.0"),
+        chainId,
+        address(this)
+      )
+    );
+    emit EIP712DomainSeparatorSet(eip712DomainSeparator);
+  }
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return The storage, major, minor, and patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  /**
+   * @notice Returns info about attestations (with unrevoked signers)
+   *   for an identifier produced by a list of issuers
+   * @param identifier Hash of the identifier
+   * @param trustedIssuers Array of n issuers whose attestations will be included
+   * @param maxAttestations Limit the number of attestations that will be returned
+   * @return for m found attestations, m <= maxAttestations:
+   *         [
+   *           Array of m accounts,
+   *           Array of m issuedOns,
+   *           Array of m signers
+   *         ]; index corresponds to the same attestation
+   * @dev Adds attestation info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+
+  // TODO ASv2 consider also returning an array with counts of attestations per issuer
+  function lookupAttestations(
+    bytes32 identifier,
+    address[] memory trustedIssuers,
+    uint256 maxAttestations
+  ) public view returns (address[] memory, uint256[] memory, address[] memory) {
+    // Cannot dynamically allocate an in-memory array
+    // For now require a max returned parameter to pre-allocate and then shrink
+    // TODO ASv2 is it a risk to allocate an array of size maxAttestations?
+    // Same index corresponds to same attestation
+    address[] memory accounts = new address[](maxAttestations);
+    uint256[] memory issuedOns = new uint256[](maxAttestations);
+    address[] memory signers = new address[](maxAttestations);
+
+    uint256 currIndex = 0;
+
+    for (uint256 i = 0; i < trustedIssuers.length; i = i.add(1)) {
+      uint256 numTrustedIssuers = identifierToAddresses[identifier][trustedIssuers[i]].length;
+      for (uint256 j = 0; j < numTrustedIssuers; j = j.add(1)) {
+        // Only create and push new attestation if we haven't hit max
+        if (currIndex < maxAttestations) {
+          // solhint-disable-next-line max-line-length
+          OwnershipAttestation memory attestation = identifierToAddresses[identifier][trustedIssuers[i]][j];
+          if (!revokedSigners[attestation.signer]) {
+            accounts[currIndex] = attestation.account;
+            issuedOns[currIndex] = attestation.issuedOn;
+            signers[currIndex] = attestation.signer;
+            currIndex = currIndex.add(1);
+          }
+        } else {
+          break;
+        }
+      }
+    }
+
+    // Trim returned structs if necessary
+    if (currIndex < maxAttestations) {
```

Comment:
Here and in the next three lines, this should be `currIndex.add(1)`.

Reply:
I don't think so, we increment `currIndex` after finding each new attestation, so after finding/appending the last attestation, it should already be at lastEltIndex + 1 = length.

double checked with the unit tests and `currIndex.add(1)` results in trimmed arrays with an additional empty element at the end

Reply:
🤯

Reply:
Nice, that makes sense 👌